tests/test_discover_latent_nodes.py

```diff
@@ -0,0 +1,175 @@
+"""This module tests the methods involved in discovering and marking latent nodes."""
+
+import unittest
+from copy import deepcopy
+
+from eliater.discover_latent_nodes import find_all_nodes_in_causal_paths, mark_latent
+from eliater.examples import ecoli, sars, t_cell_signaling_pathway
+from eliater.frontdoor_backdoor import (
+    multi_mediators_confounders_nuisance_vars_example,
+    multiple_mediators_single_confounder_example,
+)
+from y0.dsl import Variable
+from y0.graph import set_latent
+
+
+class TestDiscoverAndMarkLatentNodes(unittest.TestCase):
```

Comment:
none of the tests here are interpretable. Can you please add some explicit, minimum viable tests? This means make the smallest possible graph that exhibits the way the algorithm is supposed to work and define it explicitly within the test. Here, it requires understanding large complex graphs to understand what the tests are doing, which is also error prone during implementation

Reply:
Added a simple network